### DISCUSSION TITLE
Support new MyClass()->method() without parentheses (PHP 8.4)

### DISCUSSION THREAD

**Discussion by vudaltsov** [PHP-CS-Fixer/PHP-CS-Fixer#8069] [Feature ideas]
Hi! I would like to try to add support for my accepted [RFC](https://wiki.php.net/rfc/new_without_parentheses) that allows to omit parentheses around `new` expressions.

Where this change fits best? If I get it right, it should not be a part of the `NoUnneededControlParenthesesFixer`, because `new` is an expression, not a control structure. Is it then related to https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/7329?

**Comment by kubawerlos** [comment]
I'd say the `new_without_parentheses` from RFC's URL https://wiki.php.net/rfc/new_without_parentheses it as good name.

**Comment by vudaltsov** [comment]
So you think that it should be a separate Fixer? What about creating a fixer that removes needless parentheses from all expressions? Like `(('abc'))` -> `'abc'`.

**Comment by Wirone** [comment]
In my opinion there are several options:
- deprecate `no_unneeded_control_parentheses` in favor of `no_superfluous_parentheses` (I like this naming better than "unneeded", and we have 2 vs 4 such fixers so we need to align them anyway) and include support for `new` there. But this approach is one-way, which not necessarily is good.
- add a new fixer that would focus only on `new` initializer. But in this case I am not sure if @kubawerlos' [suggestion](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/8069#discussioncomment-9683143) is optimal, as I know several people who want to continue with parentheses even if they're not required. That's why I believe this fixer should be bi-directional, and thus can't be named `new_without_parentheses` 🙂. Especially there already is `new_with_parentheses` fixer, which handles "inner" parentheses, while this new one would focus on "outer" parentheses so the name must be more appropriate, but it's hard to find a name that wouldn't be super long and over-descriptive 😅.
- which leads to the 3rd option, which is deprecating `new_with_parentheses` in favour of `new_parentheses` and providing 2 options there: `invoke` (for `new Foo()`) and `chain` (for `(new Foo)->`). Options' names can differ, it's only an idea. Both options should be tri-state: `bool|null`, which means you can enforce parentheses (`true`), remove them (`false`) or leave them as-is (`null`).

Personally I would go with the 3rd one.

**Reply by vudaltsov** [reply]
Yes, I also like the third option.
